### PR TITLE
Added ObjectList::uniqueFor.

### PR DESCRIPTION
**Pull request by pies**

### CODE REVIEW

File: src/ObjectList.php [hunk 1]
```diff
@@ -194,6 +194,25 @@ public function exclude(ObjectList $otherList)
     }
 
     /**
+     * Returns a new list containing one element for each {$selector($element)} value (i.e. array_unique using
+     * a callback for each value).
+     *
+     * @param callable $selector
+     * @return static
+     */
+    public function uniqueFor(callable $selector)
```
Comment (OneWebPro) [inline src/ObjectList.php]: I think I don't understands that method :|

Comment (OneWebPro) [inline src/ObjectList.php]: Why just not filter?

Comment (pies) [inline src/ObjectList.php]: It's the equivalent of array_unique, not array_filter.

**Comment by OneWebPro** [comment]
Looks ok :) :+1:

**Pull request merged**